PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/localrun/FunctionRunState.java

```diff
@@ -246,18 +262,56 @@ private ProcessBuilder getRunFunctionCliProcessBuilder(File stagingFolder, int f
+        final String funcPath = functionRunConfiguration.getFuncPath();
+        String[] command = new String[]{funcPath, "extensions", "install", "--java"};
+        processBuilder.command(command);
+        processBuilder.directory(stagingFolder);
+        return processBuilder;
+    }
+
     private void prepareStagingFolder(File stagingFolder, RunProcessHandler processHandler) throws AzureExecutionException {
         ReadAction.run(() -> {
             final Path hostJsonPath = FunctionUtils.getDefaultHostJson(project);
             final Path localSettingsJson = Paths.get(functionRunConfiguration.getLocalSettingsJsonPath());
             final PsiMethod[] methods = FunctionUtils.findFunctionsByAnnotation(functionRunConfiguration.getModule());
             try {
-                FunctionUtils.prepareStagingFolder(stagingFolder.toPath(), hostJsonPath, functionRunConfiguration.getModule(), methods);
-                FunctionUtils.copyLocalSettingsToStagingFolder(stagingFolder.toPath(), localSettingsJson, functionRunConfiguration.getAppSettings());
+                Map<String, FunctionConfiguration> configMap =
+                        FunctionUtils.prepareStagingFolder(stagingFolder.toPath(), hostJsonPath, functionRunConfiguration.getModule(), methods);
+                FunctionUtils.copyLocalSettingsToStagingFolder(stagingFolder.toPath(),
+                                                                                localSettingsJson, functionRunConfiguration.getAppSettings());
+
+                final Set<BindingEnum> bindingClasses = getFunctionBindingEnums(configMap);
+                if (isInstallingExtensionNeeded(bindingClasses, processHandler)) {
+                    installProcess = getRunFunctionCliExtensionInstallProcessBuilder(stagingFolder).start();
+                }
             } catch (AzureExecutionException | IOException e) {
-                throw new AzureExecutionException("Failed to prepare staging folder", e);
+                throw new AzureExecutionException("Failed to prepare staging folder due to error: " + e.getMessage(), e);
             }
         });
+        if (installProcess != null) {
+            try {
+                readInputStreamByLines(installProcess.getErrorStream(), inputLine -> {
+                    if (processHandler.isProcessRunning()) {
+                        processHandler.println(inputLine, ProcessOutputTypes.STDERR);
+                    }
+                });
+                readInputStreamByLines(installProcess.getInputStream(), inputLine -> {
+                    if (processHandler.isProcessRunning()) {
+                        processHandler.setText(inputLine);
+                    }
+                });
+                int exitCode = installProcess.waitFor();
+                if (exitCode != 0) {
+                    throw new AzureExecutionException(INSTALL_FUNCTION_EXTENSIONS_FAIL);
+                }
+            } catch (AzureExecutionException e) {
```

Comment:
No need to catch it, we could throw it directly

Reply:
Since the next catch(Exception... will cover the AzureExecutionException, I don't want to use AzureExecutionException  to wrap AzureExecutionException  again